tests/php/Shortcodes/ImageShortcodeProviderTest.php

```diff
@@ -380,4 +482,39 @@ public function testCreateImageTag(string $expected, array $attributes)
     {
         $this->assertEquals($expected, ImageShortcodeProvider::createImageTag($attributes));
     }
+
+    /**
+     * This method will assert that the $tag will contain an image with specific attributes and values
+     *
+     * @param array $attrs Key pair values of attributes and values.
+     *                     If the value is FALSE we assume that it is not present
+     */
+    private function assertImageTag(array $attrs, string $shortcode, $message = ""): void
+    {
+        $parser = new ShortcodeParser();
+        $parser->register('image', [ImageShortcodeProvider::class, 'handle_shortcode']);
+        $tag = $parser->parse($shortcode);
+
+        $doc = new DOMDocument();
+        $doc->loadHTML($tag);
+        $node = $doc->getElementsByTagName('img')->item(0);
+        $nodeAttrs = $node->attributes;
+
+        foreach ($attrs as $key => $value) {
+            $nodeAttr = $nodeAttrs->getNamedItem($key);
+            if ($value === false) {
+                if ($nodeAttr !== null) {
+                    $this->fail("$message\nImage should not contain attribute '$key'\n$tag");
```

Comment:
Instead of doing straight assertion here, I decided to directly fail the test. This has a few advantages:
- When I was asserting something about the node it would give a very unpleasant about what was failing which made it difficult to figure out what was actually wrong.
- I'm building much more specific messages here:
  - I'm specifying what key is expected and what its expected value is.
  - I'm printing out the generated image tag.
  - I'm also including the overarching message coming from the original test case to better understand what's the actual scenario we are testing.